core/src/main/scala/pink/cozydev/lucille/QueryParser.scala

```diff
@@ -66,6 +66,8 @@ private object Parser {
     P.charIn(baseRange -- special)
   val reserved = Set("OR", "||", "AND", "&&", "NOT", "+", "-", "/")
 
+  val queryEnd = (wsp | P.end | P.char(')')).peek
```

Comment:
Curious - What is peek doing really?

Reply:
`peek` succeeds if the underlying parser would succeed but then it always backtracks afterwards, so you're "peeking" at the result but not consuming it. In our case what it means is that we're making sure a boost or a minimum match query ends with one of the those three characters, but we don't consume those characters as part of the query. This is mostly important for `)`, because the closing paren is actually part of the `group` query and if we consume it then we wouldn't be able to properly parse the group (in my head, I think of it like boost/minimum match need to know it's there, but the `)` doesn't belong to them)

Does that make sense?

Reply:
It does. Look at us - you explaining stuff to me on PRs. I've missed this! 
Thanks

Reply:
Me tooooooo ❤️❤️❤️❤️❤️❤️❤️